java-checks/src/main/java/org/sonar/java/checks/regex/UnicodeAwareCharClassesCheck.java

```diff
@@ -19,113 +19,28 @@
  */
 package org.sonar.java.checks.regex;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import org.sonar.check.Rule;
-import org.sonarsource.analyzer.commons.collections.MapBuilder;
-import org.sonarsource.analyzer.commons.collections.SetUtils;
 import org.sonar.java.regex.RegexCheck;
-import org.sonarsource.analyzer.commons.regex.RegexParseResult;
-import org.sonarsource.analyzer.commons.regex.ast.CharacterRangeTree;
-import org.sonarsource.analyzer.commons.regex.ast.EscapedCharacterClassTree;
-import org.sonarsource.analyzer.commons.regex.ast.NonCapturingGroupTree;
-import org.sonarsource.analyzer.commons.regex.ast.RegexBaseVisitor;
-import org.sonarsource.analyzer.commons.regex.ast.RegexTree;
 import org.sonar.plugins.java.api.tree.ExpressionTree;
 import org.sonar.plugins.java.api.tree.Tree;
+import org.sonarsource.analyzer.commons.regex.RegexParseResult;
+import org.sonarsource.analyzer.commons.regex.finders.UnicodeUnawareCharClassFinder;
 
 @Rule(key = "S5867")
 public class UnicodeAwareCharClassesCheck extends AbstractRegexCheck {
 
-  private static final List<Character> unicodeAwareClassesWithFlag = Arrays.asList('s', 'S', 'w', 'W');
-  private static final Set<String> unicodeAwarePropertiesWithFlag = SetUtils.immutableSetOf(
-    "Lower", "Upper", "Alpha", "Alnum", "Punct", "Graph", "Print", "Blank", "Space");
-
-  private static final Map<Character, Character> unicodeUnawareCharacterRanges = MapBuilder.<Character, Character>newMap()
-    .put('a', 'z')
-    .put('A', 'Z')
-    .build();
-
   @Override
   public void checkRegex(RegexParseResult regexForLiterals, ExpressionTree methodInvocationOrAnnotation) {
-    new UnicodeUnawareCharClassFinder(methodInvocationOrAnnotation).visit(regexForLiterals);
+    new UnicodeUnawareCharClassFinder(this::reportIssueFromCommons, (message, cost, secondaries) -> {
+      String flagName = methodInvocationOrAnnotation.is(Tree.Kind.ANNOTATION) ? "(?U)" : "UNICODE_CHARACTER_CLASS";
+      reportIssue(methodOrAnnotationName(methodInvocationOrAnnotation),
+        message.replace("\"u\"", String.format("\"%s\"", flagName)),
```

Comment:
The difference on the Java side compared to the current implementation on Commons is that we have a small logic to suggest the best flag to use. We can still apply the logic and adjust the message that we receive.